### PR TITLE
feat: Enhance Status Column with Colors for Better Readability

### LINKED ISSUE #161: Colorize the Status column in the Invoice Dashboard

### PR DESCRIPTION
**Pull request by sstefdev**
Fixes #161 

### Problem
The lack of color coding in the Status column makes it difficult for users to quickly identify the status of invoices at a glance, reducing the efficiency of the Invoice Dashboard.

### Changes 
- Added a reusable component to handle the color-coded labels for all invoice statuses.
- Incorporated the component into the Invoice Dashboard to replace the plain text status.
- Used visually distinct colors and patterns to make statuses easily recognizable.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `StatusLabel` component for improved status display across the invoice dashboard.
	- Enhanced error handling for unsupported payment networks in the invoice view.

- **Bug Fixes**
	- Improved loading state management and error status resets in payment requests.

- **Documentation**
	- Updated documentation to reflect changes in status handling and component usage.

### CODE REVIEW

**Comment by rodrigopavezi** [comment]
@sstefdev could please resolve the coderabitai reviews and the merging conflicts?

File: shared/utils/capitalize.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export const capitalize = (str: string) =>
+  (str && str[0].toUpperCase() + str.slice(1)) || "";
```
[the remark this answers is not in the document]
Comment (sstefdev) [reply shared/utils/capitalize.ts:2]: No need for this

File: shared/utils/checkStatus.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { capitalize } from "./capitalize";
+import { Types } from "@requestnetwork/request-client.js";
+
+export const checkStatus = (request: Types.IRequestDataWithEvents | null) => {
+  const balance = BigInt(request?.balance?.balance ?? 0);
+  const expectedAmount = BigInt(request?.expectedAmount ?? 0);
+  const today = new Date();
+  const dueDate = new Date(request?.contentData?.paymentTerms?.dueDate);
+  const isPaid = balance >= expectedAmount ? "Paid" : "Partially Paid";
```
[the remark this answers is not in the document]
Comment (sstefdev) [reply shared/utils/checkStatus.ts:9]: This is added in another PR

File: shared/utils/checkStatus.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      return "Overdue";
+    }
+    return isPaid;
+  } else {
+    if (balance === BigInt(0)) {
+      return "Awaiting Payment";
+    }
+    return isPaid;
+  }
+};
```
[the remark this answers is not in the document]
Comment (sstefdev) [reply shared/utils/checkStatus.ts:37]: This is also added

File: shared/components/status-label.svelte [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      case "overdue":
+        statusClass = "status-overdue";
+        break;
+      case "pending":
+        statusClass = "status-pending";
+        break;
+      default:
+        statusClass = "status-created";
+    }
+  }
```
[the remark this answers is not in the document]
Comment (sstefdev) [reply shared/components/status-label.svelte:37]: there is always a status of created

**Review by aimensahnoun — APPROVED** [review]

**Pull request merged**